[Windows|macOS|Linux|BSD] - FTBFS on GCC 14

> #### Checklist
> 
> <!-- You don't have to fill these out, but it would help us alot with managing open bug reports if you make sure to check these things first! -->
> 
> - [X] I am reporting exactly 1 bug with this issue.
> - [X] This bug hasn't already been reported.
> - [X] This bug hasn't already been fixed in the latest development build.

---

## Bug Description

I am trying to build Bambootracker with gcc-14.0.1 on Fedora 40.
The build fails with the following error:
```
gcc -c -pipe -std=gnu11 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wall -Wextra -Wall -Wextra -Werror -pedantic -pedantic-errors -D_REENTRANT -fPIC -DQT_DEPRECATED_WARNINGS -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I. -Iinstrument -Imodule -I../submodules/emu2149/src -I../submodules/RtAudio/src -I../submodules/RtMidi/src -I/usr/include/qt5 -I/usr/include/qt5/QtWidgets -I/usr/include/qt5/QtGui -I/usr/include/qt5/QtCore -I. -I. -I/usr/lib64/qt5/mkspecs/linux-g++ -o wopn_file.o format/wopn_file.c
make[1]: Leaving directory '/builddir/build/BUILD/BambooTracker/BambooTracker'
format/wopn_file.c: In function â€˜WOPN_SaveBankToMemâ€™:
format/wopn_file.c:555:17: error: â€˜strncpyâ€™ output may be truncated copying 32 bytes from a string of length 32 [-Werror=stringop-truncation]
  555 |                 strncpy((char*)cursor, bankslots[i][j].bank_name, 32);
      |                 ^
cc1: all warnings being treated as errors
```

## System Information

<!-- Please fill out these details about your system, to help us debug your problem. -->

- **Operating System**: Linux Fedora 40
- **BambooTracker Version**: 0.6.3
- **Build Type**: local build

--- a/BambooTracker/format/wopn_file.c
+++ b/BambooTracker/format/wopn_file.c
@@ -552,7 +552,7 @@ int WOPN_SaveBankToMem(WOPNFile *file, void *dest_mem, size_t length, uint16_t v
             {
                 if(length < 34)
                     return WOPN_ERR_UNEXPECTED_ENDING;
-                strncpy((char*)cursor, bankslots[i][j].bank_name, 32);
+                memcpy(cursor, bankslots[i][j].bank_name, 32);
                 cursor[32] = bankslots[i][j].bank_midi_lsb;
                 cursor[33] = bankslots[i][j].bank_midi_msb;
                 GO_FORWARD(34);

